implementations/micrometer-registry-prometheus/src/main/java/io/micrometer/prometheus/PrometheusNamingConvention.java

```diff
@@ -53,20 +53,12 @@ public PrometheusNamingConvention(String timerSuffix) {
     public String name(String name, Meter.Type type, @Nullable String baseUnit) {
         String conventionName = NamingConvention.snakeCase.name(name, type, baseUnit);
 
-        switch (type) {
-            case COUNTER:
-            case DISTRIBUTION_SUMMARY:
-            case GAUGE:
-                if (baseUnit != null && !conventionName.endsWith("_" + baseUnit))
-                    conventionName += "_" + baseUnit;
-                break;
-        }
-
         switch (type) {
             case COUNTER:
                 if (!conventionName.endsWith("_total"))
                     conventionName += "_total";
                 break;
+            case DISTRIBUTION_SUMMARY:
```

Comment:
For `DISTRIBUTION_SUMMARY` it would do

```java
if (baseUnit != null && !conventionName.endsWith("_" + baseUnit))
                    conventionName += "_" + baseUnit;
```

and now it does 

```java
 if (!timerSuffix.isEmpty() && conventionName.endsWith(timerSuffix)) {
                    conventionName += timerSuffix + "_seconds";
                }
```